bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ProjectEncodingPreferenceChangeListener.java

```diff
@@ -0,0 +1,198 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Ingo Mohr - Issue #166 - Add Preference to Turn Off Warning-Check for Project Specific Encoding
+ *******************************************************************************/
+package org.eclipse.core.internal.resources;
+
+import java.util.*;
+import org.eclipse.core.internal.utils.Messages;
+import org.eclipse.core.resources.*;
+import org.eclipse.core.runtime.*;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences.IPreferenceChangeListener;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences.PreferenceChangeEvent;
+import org.eclipse.core.runtime.preferences.IScopeContext;
+import org.osgi.service.component.annotations.*;
+
+@Component(service = {})
+public class ProjectEncodingPreferenceChangeListener implements IPreferenceChangeListener {
+
+	private static final String MARKER_TYPE = ValidateProjectEncoding.MARKER_TYPE;
+
+	@Reference
+	private IWorkspace workspace;
```

Comment:
I'm no expert on how this serviced stuff works, but isn't the point here that the workspace value is injected?  I.e., I don't think we need to assign it below.

Reply:
Yes this is correct, also please pass the workspace to the workspace job.

Reply:
removed the assignment